src/ws/login.html

```diff
@@ -25,8 +25,9 @@
         <div id="login" class="login-area" style="visibility: hidden;">
           <div role="form">
 
-            <div id="error-group" class="alert alert-danger" hidden>
-              <span id="login-error-message"></span>
+            <div id="error-group" class="pf-c-alert pf-m-danger pf-m-inline dialog-error group-hidden" aria-label="inline danger alert">
+              <svg style="vertical-align:-.125em" fill="currentColor" viewBox="0 0 512 512" aria-hidden="true"><path d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"/></svg>
```

Comment:
Hah, nice! I originally thought of having inline SVG in the CSS (like the other PF icons), but this seems fine, too.

The `style=` here smells like violating the C-S-P. But perhaps we just don't have that on the login page? Or perhaps it's silently discarded? For cleanliness, could this use a class and the v-align goes into login.css?

Reply:
good point. I haven't noticed this since @garrett helped with that. @garrett is there some reason why it is like this or can it be moved into css?

Reply:
The login page has a lot of inline style stuff. I did not make this, even though I'm blamed for it. I just restructured everything to make it _look_ like PF4 and didn't want to overhaul the JS and tests.

But it would be very, very nice to remove all inline styles on the login page.

Reply:
I didn't check for other inline styles, sorry. If that's the case, then this is moot of course. Let's just drop the translatable= flag then.

Reply:
@martinpitt SVGs are so flexible. Include them with `<embed>`, `<picture>`, `<object>`, `<img>` (use img when possible; the rest are mainly deprecated ways to include them, IIRC), inline in CSS, inline in HTML, generated by JS, and so on.

In most cases, the best option is as a separate file and treat them like stretchable images. Having them inline in HTML lets you recolor them more easily, however. But they have to be really simple, as they bloat the HTML page and aren't usually cachable (unless it's a static page).

Having them inline here fits our use case for the log in page quite well.

Also, I originally thought of inlining them into our CSS too... but realized that we wouldn't have to encode them in the HTML and it was a better fit. :wink:

Reply:
Heh, one more thing - acording to Firefox this style is also unused, because "vertical-align has no effect on this element since it's not an inline or table-cell element. Try adding display:inline or display:table-cell". So I guess we can just drop it :)

Reply:
Does something change this element to inline or a table-cell somewhere else? If not, then, yeah, agreed; it can go.